src/services/document-service.spec.js

```diff
@@ -1038,4 +1048,114 @@ describe('document-service', () => {
       });
     });
   });
+
+  describe('createRegenerateDocumentsBatch', () => {
+    let createdDocument;
+    let result;
+    let revision;
+    beforeEach(async () => {
+      revision = {
+        title: 'Title',
+        slug: 'my-doc',
+        language: 'en',
+        sections: [],
+        tags: ['tag-1']
+      };
+      result = await sut.createNewDocumentRevision({ doc: revision, user });
+      createdDocument = await db.documents.findOne({ key: result.key });
+    });
+
+    describe('when a running batch already exists', () => {
+      beforeEach(async () => {
+        await sut.createRegenerateDocumentsBatch(user);
+      });
+
+      it('should return null', async () => {
+        const batch = await sut.createRegenerateDocumentsBatch(user);
+        expect(batch).toBeNull();
+      });
+    });
+
+    describe('when a new batch is created', () => {
+      let batch = null;
+      beforeEach(async () => {
+        batch = await sut.createRegenerateDocumentsBatch(user);
+      });
+
+      it('should create the new batch', () => {
+        expect(batch).toEqual({
+          _id: expect.stringMatching(/\w+/),
+          createdBy: user._id,
+          createdOn: new Date(),
```

Comment:
maybe `createdOn: now,` makes it clearer to the reader that we're using a fake timer

Reply:
Fixed, thanks!